index.d.ts

```diff
@@ -0,0 +1,78 @@
+// Type definitions for Elastic APM Node.js Agent 1.x
+// Project: https://www.elastic.co/solutions/apm
+// Definitions by: Shahaed Hasan <https://github.com/shahaed>
+// Definitions: https://github.com/DefinitelyTyped/DefinitelyTyped
+// TypeScript Version: 2.7
+
+export = a;
+
+declare const a: a.Agent;
+
+declare namespace a {
+
+    interface Agent {
+
+        currentTransaction: any;
+        logger: any;
+
+        start(opts?: AgentOptions): Agent;
+        isStarted(): boolean;
+        addFilter(filter: (payload: any) => any): void;
+        setUserContext(context: Context): boolean;
+        setCustomContext(context: any): boolean;
+        setTag(name: string, value: string): any;
+        addTags(tags: any): any;
+        captureError(error: Error | string | object, options?: object, callback?: (err?: any) => any): void;
+        startTransaction(name?: string, type?: string): any;
+        endTransaction(result?: any): any;
+        setTransactionName(name: string): any;
+        startSpan(name?: string, type?: any): any;
```

Comment:
It would be good to implement and return a `Span` type here. Also, `type` should be `string`.